src/xenia/kernel/xfile.cc

```diff
@@ -122,14 +122,12 @@ X_STATUS XFile::Read(uint32_t buffer_guest_address, uint32_t buffer_length,
       const xe::BaseHeap* buffer_end_heap =
           memory()->LookupHeap(buffer_guest_high_address);
       if (!buffer_start_heap || !buffer_end_heap ||
-          buffer_start_heap->IsGuestPhysicalHeap() !=
-              buffer_end_heap->IsGuestPhysicalHeap() ||
-          (buffer_start_heap->IsGuestPhysicalHeap() &&
+          (buffer_start_heap->heap_type() == HeapType::kGuestPhysical &&
```

Comment:
What check that does that too were you referring to in DMs? This is for cases like reading 16 bytes to 0x9FFFFFF8, where the read crosses the boundary. Without this check, this read will be treated as just reading to virtual memory.